### ISSUE TITLE
Cannot get the benchmark result using KPConv on S3DIS dataset.

### ISSUE THREAD

**Issue by LuZaiJiaoXiaL** [isl-org/Open3D-ML#526]
### Checklist

- [X] I have searched for [similar issues](https://github.com/isl-org/Open3D-ML/issues).
- [X] I have tested with the [latest development wheel](http://www.open3d.org/docs/latest/getting_started.html#development-version-pip).
- [X] I have checked the [release documentation](http://www.open3d.org/docs/release/) and the [latest documentation](http://www.open3d.org/docs/latest/) (for `master` branch).

### My Question

Using RandLA-Net on S3DIS with the default randlanet_s3dis.yml can get the benchmark result.
But I cannot get the ideal mIoU result when using KPConv with the default config file on the same S3DIS dataset. The mIoU fluctuated at 10%.

This is my code for training.

from ml3d.torch.models import kpconv

from ml3d.torch.pipelines import semantic_segmentation

from ml3d.datasets.s3dis import *

import torch

import open3d.ml as _ml3d

import sys, os

if __name__ == '__main__':
    os.chdir(sys.path[0])

    cfg_file = "./ml3d/configs/kpconv_s3dis.yml"
    cfg = _ml3d.utils.Config.load_from_file(cfg_file)

    dataset = S3DIS(**cfg.dataset)

    model=kpconv.KPFCNN(**cfg.model)

    pipeline = semantic_segmentation.SemanticSegmentation(model=model, dataset=dataset,**cfg.pipeline)

    pipeline.run_train()

Can anyone get the reported result? I would appreciate it if you can give me some suggestions.

Cheers,

Eric.

**Comment by shayan-nikoo** [comment]
On which S3DIS dataset are you running the training? The original Kp-Conv uses not-aligend version, see here: https://github.com/HuguesTHOMAS/KPConv-PyTorch/blob/master/doc/scene_segmentation_guide.md
And can you share your results log for the last several epochs?

**Comment by Arno-Tu** [comment]
Hello, I saw that you mentioned in your question that you successfully reproduced the RandLA-Net network on S3DIS dataset. I would like to ask what are the network parameters you set? When I recently trained S3DIS data with RandLA-Net, I got all the loses value as NaN

**Comment by yfq512** [comment]
@maosuli @shayan-nikoo @Arno-Tu 
I find you successfully reproduced the RandLA-Net network on S3DIS dataset, Do you know how to manually perform data preprocessing for s3dis?

**Comment by Arno-Tu** [comment]
@yfq512 The preprocessing code for S3DIS datasets is in the Open3D library. If you don't have any additional processing methods, you can call it through the following code:
`dataset = ml3d.datasets.S3DIS(cfg.dataset.pop('dataset_path', None), **cfg.dataset)`
If you want to do some extra operations during the preprocessing, then you can go to Open3D. dataset. S3DIS to see how it is handled and make some fine tuning

**Comment by yfq512** [comment]
@Arno-Tu Thanks for your reply.

I briefly reviewed the [Open3D-ML/blob/main/ml3d/datasets/s3dis.py](https://github.com/isl-org/Open3D-ML/blob/main/ml3d/datasets/s3dis.py) file and found that much of it involves operations based on .pkl and .npy format files. However, the S3DIS data I currently have only consists of some txt format files. How should I generate these pkl and npy files? I couldn't find any code in the Open3D-ML library related to S3DIS data preprocessing, like [Open3D-ML/blob/main/scripts/preprocess_scannet.py](https://github.com/isl-org/Open3D-ML/blob/main/scripts/preprocess_scannet.py). Or are you saying that just by using the 
```
dataset = ml3d.datasets.S3DIS(cfg.dataset.pop('dataset_path', None), **cfg.dataset)
```` 
line of code, the generation of pkl and npy files is automatically completed?


My S3DIS dataset directory is this:
```
datasets/S3DIS
└── Stanford3dDataset_v1.2_Aligned_Version
    ├── Area_1
    │   ├── pantry_1
    │   │   ├── Annotations
    │   │   │   ├── beam_1.txt
    │   │   │   ├── beam_2.txt
……
    │   │   └── pantry_1.txt
    │   └── WC_1
    │       ├── Annotations
    │       │   ├── ceiling_1.txt
……
    │       └── WC_1.txt
    ├── Area_2
    │   ├── pantry_1
    │   │   ├── Annotations
    │   │   │   ├── beam_1.txt
    │   │   │   ├── beam_2.txt
  ……
    │   │   └── pantry_1.txt
    │   └── WC_1
    │       ├── Annotations
    │       │   ├── ceiling_1.txt
    │       │   ├── ceiling_2.txt
    │       │   ├── clutter_10.txt


```

**Comment by Arno-Tu** [comment]
@yfq512 Yes, you are right. The preprocessing process of S3DIS is to read xyz and rgb information from. txt, generate a label based on its file name, finally combine them, and then generate. pkl and. npy files. I remember that the specific code executed is the create _ ply _ files function in s3dis.py, you can take a look.

**Comment by yfq512** [comment]
@Arno-Tu Thank you. You're such a kind person. My code is worked.🌹

**Comment by Arno-Tu** [comment]
@yfq512 you're welcome ! I be glad to can help you solve your problem.
